modules/survey_module/jsx/DirectEntry.js

```diff
@@ -0,0 +1,384 @@
+/**
+ * This file contains React component for Direct Data Entry
+ *
+ * @author Jordan Stirling (StiringApps ltd.)
+ * @version 0.0.1
+ *
+ */
+
+/**
+ * Panel component
+ * Wraps children in a collapsible bootstrap panel
+ */
+
+import Page from './DirectEntryForm';
+
+/*eslint-disable */
+class DirectEntry extends React.Component {
+
+	constructor(props) {
+	    super(props);
+
+	    const height = window.innerHeight / 3;
+	    const style = {
+	    	"margin-top": height
+	    }
+	    let page = -1;
+		let url = window.location;
+		if (!window.location.origin) {
+  			url.origin = url.protocol
+  				+ "//" + url.hostname
+  				+ (url.port ? ':' + url.port: '');
+		}
+
+	    this.nextPage = this.nextPage.bind(this);
+	    this.prevPage = this.prevPage.bind(this);
+	    this.updateAnswer = this.updateAnswer.bind(this);
+	    this.setupPageValues = this.setupPageValues.bind(this);
+	    this.submit = this.submit.bind(this);
+
+	    this.state = {
+	    	style: style,
+	    	page: page,
+	    	values: {},
+	    	errors: {},
+	    	InstrumentJSON: {},
+	    	completionStats: {
+				total: 0,
+				completed: 0
+			},
+			api_url: url.origin + "/survey_module/ajax/survey_api.php" + url.search
+	    };
+
+	}
+
+	componentWillMount() {
+
+		$.ajax({
```

Comment:
This should be done in ComponentDidMount because ComponentWillMoiunt is done just before the first render and this.setState might not trigger a second rendering whit the new state. Otherwise, the method name is now UNSAFE_componentWillMount()